packages/ui/src/components/primitives/AccessibleList.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { forwardRef, type FC } from 'react'
+import * as ToggleGroup from '@radix-ui/react-toggle-group'
+import {
+  cva,
+  css as designCss,
+  type Styles
+} from '@reservoir0x/relay-design-system/css'
+
+const AccessibleListContainerCss = cva({
+  base: {
+    display: 'flex',
+    flexDirection: 'column'
+  }
+})
+
+const AccessibleListItemCss = cva({
+  base: {
+    display: 'flex',
+    alignItems: 'center',
+    position: 'relative',
+    userSelect: 'none',
+    cursor: 'pointer',
+    '&[data-state=on]': {}
+  }
+})
+
+type AccessibleListProps = {
+  children: React.ReactNode
+  onSelect: (value: string) => void
+  css?: Styles
+}
+
+export const AccessibleList: FC<AccessibleListProps> = ({
+  children,
+  onSelect,
+  css
+}) => {
+  return (
+    <ToggleGroup.Root
+      type="single"
+      loop={false}
+      onValueChange={onSelect}
+      className={designCss(
+        AccessibleListContainerCss.raw(),
+        designCss.raw(css)
+      )}
+    >
+      {children}
+    </ToggleGroup.Root>
+  )
+}
+
+type ListItemProps = {
+  children: React.ReactNode
+  value: string
+  css?: Styles
+  asChild?: boolean
+}
+
+export const ListItem = forwardRef<HTMLButtonElement, ListItemProps>(
```

Comment:
I feel like this should be named: AccessibleListItem, so the hierarchy is clear

Reply:
Will do!